lib/interceptors/csrf-token.ts

```diff
@@ -0,0 +1,29 @@
+import { generateUrl } from '@nextcloud/router'
+
+const RETRY_KEY = Symbol('csrf-retry')
+
+export const onError = axios => async (error) => {
+	const { config, response, request: { responseURL } } = error
+	const { status } = response
+
+	if (status === 412
+		&& response?.data?.message === 'CSRF check failed'
+		&& config[RETRY_KEY] === undefined) {
+		console.warn(`Request to ${responseURL} failed because of a CSRF mismatch. Fetching a new token`)
+
+		const { data: { token } } = await axios.get(generateUrl('/csrftoken'))
+		console.debug(`New request token ${token} fetched`)
```

Comment:
**Question:** It is secure to do a log about this token?

Reply:
Yes, as it is delivered with html pages anyways. It should never leave the session but it doesn't contain any secrets.

Reply:
PS: The heartbeat also prints the token, IIRC.